core-bundle/src/Resources/contao/classes/DataContainer.php

```
@@ -1541,6 +1541,106 @@ public static function getDriverForTable(string $table): string
 
 		return $dataContainer;
 	}
+
+	/**
+	 * Generates the label for a given data record according to the DCA configuration.
+	 *
+	 * @param array  $row   The data record
+	 * @param string $table The name of the data container
+	 */
+	public static function generateRecordLabel($row, $table): string
+	{
+		$showFields = $GLOBALS['TL_DCA'][$table]['list']['label']['fields'] ?? array('id');
```

I don't think we should fall back to the ID. How does the current implementation behave if the fields are not defined? Maybe throw an exception that either fields or label_callback must be defined?